share-coordinator/src/test/java/org/apache/kafka/coordinator/share/SharePartitionOffsetManagerTest.java

```diff
@@ -0,0 +1,205 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.coordinator.share;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.server.share.SharePartitionKey;
+import org.apache.kafka.timeline.SnapshotRegistry;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SharePartitionOffsetManagerTest {
+
+    private SharePartitionOffsetManager manager;
+    private static final SharePartitionKey KEY1 = SharePartitionKey.getInstance("gs1", Uuid.randomUuid(), 0);
+    private static final SharePartitionKey KEY2 = SharePartitionKey.getInstance("gs2", Uuid.randomUuid(), 0);
+    private static final SharePartitionKey KEY3 = SharePartitionKey.getInstance("gs1", Uuid.randomUuid(), 1);
+    private static final SharePartitionKey KEY4 = SharePartitionKey.getInstance("gs1", Uuid.randomUuid(), 7);
+
+    @BeforeEach
+    public void setUp() {
+        SnapshotRegistry snapshotRegistry = new SnapshotRegistry(new LogContext(SharePartitionOffsetManagerTest.class.getName()));
+        manager = new SharePartitionOffsetManager(snapshotRegistry);
+    }
+
+
+    @Test
+    public void testNullRegistryArgInvocation() {
+        assertThrows(NullPointerException.class, () -> new SharePartitionOffsetManager(null));
+    }
+
+    @Test
+    public void testUpdateStateAddsToInternalState() {
+        assertEquals(Optional.empty(), manager.updateState(KEY1, 0L));
+        assertEquals(Optional.of(10L), manager.updateState(KEY1, 10L));
+        assertEquals(Optional.empty(), manager.updateState(KEY2, 15L));
+
+        assertEquals(10L, manager.offsets().get(KEY1));
+        assertEquals(15L, manager.offsets().get(KEY2));
+    }
+
+    private static class ShareOffsetTestHolder {
+        static class TestTuple {
+            final SharePartitionKey key;
+            final long offset;
+            final Optional<Long> expectedOffset;
+
+            private TestTuple(SharePartitionKey key, long offset, Optional<Long> expectedOffset) {
+                this.key = key;
+                this.offset = offset;
+                this.expectedOffset = expectedOffset;
+            }
+
+            static TestTuple instance(SharePartitionKey key, long offset, Optional<Long> expectedOffset) {
+                return new TestTuple(key, offset, expectedOffset);
+            }
+        }
+
+        private final String testName;
+        private final List<TestTuple> tuples;
+        private final boolean shouldRun;
```

Comment:
This seems unnecessary at the moment.

Reply:
It makes very easy to run a single test among the stream. Helpful in debugging